.idea/Working version/algorithms.js

```diff
@@ -0,0 +1,472 @@
+/**
+ * Created by tsodergren on 6/5/17.
+ */
+
+
+function constructEdges() {
+
+
+    var radiusChange = arguments.length ? arguments[0] : 0;
+
+    //squared coverage diameter and annulus of uncertainty
+    var sqDiameter = 4 * Math.pow(complexRadius, 2);
+    var sqDiameterMin = 4 * Math.pow(complexRadius-dataRadius, 2);
+    var sqDiameterMax = 4 * Math.pow(complexRadius+dataRadius, 2);
+
+
+    if (!a_edges) {
+        var x=math.zeros(numSamples, numSamples,'sparse');
+        a_edges = {edgeProb: x.clone(), edgeAnchorDist: x.clone(), edgeDist: x.clone(), edgeFlag: x.clone(),
+            edgeInd: x.clone() };
+    }
+
+    var iEdgeDist, iEdgeFlag, tempEdges;
+
+    if ( radiusChange==1 ) {
+        tempEdges = JSON.parse(JSON.stringify(ripsEdges));
```

Comment:
It looks like object assign works only on the objects themselves, not arrays. I would still have to loop over the array using map or forEach. The approach I used seems to be the most recommended way to do this. https://stackoverflow.com/questions/41273735/how-do-i-copy-an-array-of-arrays-of-objects-and-be-able-to-manipulate-the-copied

As for compatibility, I know we never looked at it, but ultimately I would say we want cross-browser compatibility so probably best not to implement new functions that we know won't work on IE. I will worry about that part.